hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -82,11 +81,12 @@ class IncrementalRelation(val sqlContext: SQLContext,
     optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME_OPT_KEY, lastInstant.getTimestamp))
     .getInstants.iterator().toList
 
-  // use schema from a file produced in the latest instant
-  val latestSchema: StructType = {
+  // use schema from a file produced in the end instant
+  val endInstantSchema: StructType = {
```

Comment:
I think we should introduce a new ReadOption to control this? and default to existing behavior of using the latest schema.

Reply:
i think  add a ReadOption to control this make sense now. But as the Time Traveling feature, not only data , but also schema should use the end version.  I also want to know about Hudi’s plan on time travel ?
thanks